### PR TITLE
CFE-2390 Add body agent control select_end_match_eof option.

### PR DESCRIPTION
**Pull request by pasinskim**
New body agent control option has been added to set the behavior
of default select_end_match_eof edit_line promise constraint.

Changelog: title

### CODE REVIEW

**Comment by pasinskim** [comment]
trigger build!

**Comment by pasinskim** [comment]
@kacf can you take a look at this one? The build is green.

File: libpromises/eval_context.c [hunk 1]
```diff
@@ -129,11 +129,24 @@ struct EvalContext_
...
     /* List if all classes set during policy evaluation */
     StringSet *all_classes;
 };
 
+bool GetSelectEndMatchEof(const EvalContext *ctx)
+{
+    return ctx->select_end_match_eof;
+}
+
+void SetSelectEndMatchEof(EvalContext *ctx, bool value)
```
Comment (kacf) [inline libpromises/eval_context.c]: Most functions regarding `EvalContext` are prefixed with `EvalContext`. We should probably do that here as well.

**Comment by kacf** [comment]
Looks good otherwise. Don't forget documentation as well!

**Comment by pasinskim** [comment]
Thanks @kacf. Fixed!

**Comment by kacf** [comment]
Great!

99% sure this will work, but I'll trigger a build. I'm waiting for the dependencies to be ready anyway.

**Comment by kacf** [comment]
trigger build

**Comment by kacf** [comment]
Cherry-picked to 3.9.x as well.

**Pull request merged**